client/src/shared/codeEditor/codeTokenizer/tokenizeLine/index.ts

```diff
@@ -81,23 +91,63 @@ const tokenizeLine = (
     if (unknownUnit) {
       return createTokenizedLineWithError({
         values,
+        customFunctions,
         lineString,
         errorMessage: `Unknown unit "${unknownUnit.unit}"`,
         start: lineString.lastIndexOf('=') + 1,
       });
     }
   }
 
+  if (isFunction) {
+    return {
+      values: {},
+      customFunctions: {
+        [functionName as string]: (...args) => {
+          const { result, error } = evaluateExpression(
+            expression,
+            {
+              ...values,
+              ...(functionArguments as string[]).reduce((acc, argumentName, index) => ({
+                ...acc,
+                [argumentName]: args[index],
+              }), {}),
+            },
+            {
+              ...functions,
+              ...customFunctions,
+            },
+            unitsMap
+          );
+          if (error) {
+            throw new EvaluationError(`Error in function "${functionName}": ${error}`);
```

Comment:
It would be awesome if we could somehow perform most of the parsing/validation during declaration of a function instead of during execution. It would make error messages easier to understand and we would keep highlighting of the incorrect part of the expression. However, it doesn't seem trivial to do, so I'm fine with this approach for now.